GTBitsOfGood/southface
# rd/65 - logOutAndErrorHandling

## Logout Functionality + Error Handling

Issue Number(s): #65 
What does this PR change and why?
Added a isAdmin cookie. This cookie is used on the frontend to prevent any non Admin users to click on any of the buttons that are only allowed for Admins. The logout button now replaces the login button whenever you login/register, and then disappears after you log out.  Removed all error handling logic from the backend code related to the issue and simplifying the backend code correspondingly. Added a wrapper function in useUser to check if User is logged in and is an admin.

### Checklist

- [x] Logout button is rendered if the user is logged in (check cookies)
- [x] OnClick of logout, cookies are cleared and the login button is shown again
- [x] Error toast logic is handled purely in the frontend (see the useUser hook), no API calls are made if the user is not admin (check other notes).
- [x] Two cases for error handling is handled:
        - the user is not logged in (check if cookies are set)
        - the user is logged in, but is not admin (check if isAdmin property is true, maybe set this as a cookie when the user logs in?)

### How to Test

- Login/Register to see the logout button
- Test toast logic without login
- Login in with non Admin account to the the toast logic whenever you click on any buttons that are limited to only Admins

### Other Notes
- Right now, the logout Button is still a NavLink componenet I was thinking of spending some time this week to figure out how to change that so it is simply just kinda a button that just logs you out. It works but there is just a lot of redundant code there
- Not sure if I implemented the routes section correctly. Wasn't sure what it meant by "auth" before so I just added a login variable to filter between the logout and login button. Can change if it isn't intuitive.
- I deleted all the handleError code from the backend since we are no longer catching any specific error. Not sure if we needed to keep it, but can add it back if we do.
- I force the window to reload after logging out. This is due to the fact that the login button doesn't show up until the window is refreshed. The cookies get cleared, but the UI doesn't change accordingly. Let me know if there is a better way to do this.
